executor/prepared.go

```diff
@@ -341,10 +341,18 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 		sc.OverflowAsWarning = false
 		sc.TruncateAsWarning = !sessVars.StrictSQLMode
 		sc.InUpdateOrDeleteStmt = true
+		if node, ok := s.(*ast.UpdateStmt); ok {
+			sc.IgnoreError = node.IgnoreErr
+		} else {
+			node, _ := s.(*ast.DeleteStmt)
+			sc.IgnoreError = node.IgnoreErr
```

Comment:
when IgnoreErr is true, 
what will happen for truncateErr and overflowErr in insert, update, delete?

Reply:
Generate warnings instead of errors.

Reply:
1. According to https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#ignore-strict-comparison
we may **not** need an IgnoreErr flag in StatementContext,
we can check node.IgnoreErr here, and set the IgnoreTruncate/ TruncateAsWarning to the certain value,
2. We may need **IgnoreDividedByZero and DividedByZero** flags in StatementContext since TiDB does not support a full implementation of SQL_MODE now.

Reply:
@lkk2003rty